### ISSUE TITLE
Rust Cache warnings for tests with more than one feature enabled

### ISSUE THREAD

**Issue by frazar** [google/zerocopy#43]
Looking at [this CI log](https://github.com/google/zerocopy/actions/runs/3222541405/jobs/5271716415#step:4:30)
```
Warning:  ValidationError: Key Validation Error: v0-rust-1.56.1-i686-unknown-linux-gnu-alloc,simd--build_test-a732c1bed0bf174cd2e9b37132d29ed1815eb7cd-df69c652838aa8f1d3f759765b37bdf011baf599 cannot contain commas.
    at checkKey (/home/runner/work/_actions/Swatinem/rust-cache/v2.0.0/dist/restore/index.js:58:15)
    at Object.<anonymous> (/home/runner/work/_actions/Swatinem/rust-cache/v2.0.0/dist/restore/index.js:90:13)
    at Generator.next (<anonymous>)
    at /home/runner/work/_actions/Swatinem/rust-cache/v2.0.0/dist/restore/index.js:15:71
    at new Promise (<anonymous>)
    at __webpack_modules__.7799.__awaiter (/home/runner/work/_actions/Swatinem/rust-cache/v2.0.0/dist/restore/index.js:11:12)
    at Object.restoreCache (/home/runner/work/_actions/Swatinem/rust-cache/v2.0.0/dist/restore/index.js:80:12)
    at run (/home/runner/work/_actions/Swatinem/rust-cache/v2.0.0/dist/restore/index.js:62066:40)
```
it seems like Rust Cache does not like commas in keys.

This occurs because keys [defined as](https://github.com/google/zerocopy/blob/6aea74e78a85f7e8227d485f7e11ae4957329120/.github/workflows/ci.yml#L49)
```
key: "${{ matrix.channel }}-${{ matrix.target }}-${{ matrix.features }}-${{ hashFiles('**/Cargo.lock') }}"
```

and they end up containing commas when `${{ matrix.features }}` expands to two or more features, like "alloc,simd,simd-nightly".

Maybe the `key` definition should be changed to a comma-less verison. Also, a way to turn RustCache warnings into hard errors should be considered.

**Comment by joshlf** [comment]
Thanks for catching this! Definitely should fix, although I'm a bit on the fence about making RustCache warnings hard errors - the tests still test what we want them to, they just run somewhat more slowly. I wonder if there's a middle ground where we can make sure we're notified of failures but they don't block the rest of the tests from running, and they don't block PRs from being merged?

**Comment by joshlf** [comment]
Closed in 230820af286343c9795e0c964a8e1a15d3094d1e.
